### PR TITLE
feat: generate the monthly invoices for the BillingSystem = 'QuickBookUsa' and send them to SAP US

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Create a new job to generate the monthly invoices for the BillingSystem = **'QuickBookUsa'** and send them to SAP US.

**Task:** [DAT-188](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-188)

### CODE REVIEW

File: DopplerBillingJob/Database/DopplerRepository.cs [hunk 1]
```diff
@@ -4,40 +4,34 @@
...
-            IOptionsMonitor<DopplerBillingJobSettings> dopplerBillingJobSettings,
             IDbConnectionFactory dbConnectionFactory)
         {
             _logger = dopplerRepositoryLogger;
-            _dopplerBillingJobSettings = dopplerBillingJobSettings;
             _dbConnectionFactory = dbConnectionFactory;
         }
 
-        public async Task<IList<UserBilling>> GetUserBillingInformation()
+        public async Task<IList<UserBilling>> GetUserBillingInformation(List<string> storedProcedures)
```
Comment (andresmoschini) [inline DopplerBillingJob/Database/DopplerRepository.cs:25]: Why did you refactor it moving an object level injection to a method level injection?

Comment (jhoffman-ms) [reply DopplerBillingJob/Database/DopplerRepository.cs:25]: My idea was that the each billing job gets the stored procedures that it need to runs and send them to the repository and remove the logic  to determinate witch stored procedures need to runs depending the job in the repository. What do you think?

Comment (andresmoschini) [reply DopplerBillingJob/Database/DopplerRepository.cs:25]: So, in the future, you are planning to have different implementations of the current `DopplerBillingJob` and only one `IDopplerRepository`?

Comment (jhoffman-ms) [reply DopplerBillingJob/Database/DopplerRepository.cs:25]: Currently we have 2 Billing jobs : ```DopplerBillingJob``` and ```DopplerBillingUsJob``` and both use the same Repository. Each job pass the corresponding stored procedures:

**DopplerBillingJob:** (SAP AR)

```var billingData = await _dopplerRepository.GetUserBillingInformation(_dopplerBillingJobSettings.CurrentValue.StoredProcedures);```

**DopplerBillingUsJob:** (SAP US)

```var billingData = await _dopplerRepository.GetUserBillingInformation(_dopplerBillingUsJobSettings.CurrentValue.StoredProcedures);```

Let me know if you want a meeting to explain you with more details.

Comment (andresmoschini) [reply DopplerBillingJob/Database/DopplerRepository.cs:25]: No, I think that I am understanding, it makes sense.

**Comment by nreal** [comment]
👍

**Pull request merged**